### PR TITLE
Reduce memory usage of different pods

### PR DESCRIPTION
**Pull request by Filter94**
Signed-off-by: Ameziane H <ameziane.hamlat@consensys.net>

This PR implements issue(s) #

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] I have informed the team of any breaking changes if there are any.

### CODE REVIEW

**Review by julien-marchand — APPROVED** [review]

**Review by ahamlat — APPROVED** [review]
Let's start with this configuration and see if the suggested -Xmx are sufficient. We can also add -XX:+HeapDumpOnOutOfMemoryError `-XX:HeapDumpPath=/path-to-store-heap-dumps/heapdump-{name-of-the-pod}.hprof`. This will generate heap dumps on OOM errors.

File: docker/compose.yml [hunk 1]
```diff
@@ -141,11 +141,12 @@ services:
     restart: "no"
     environment:
       LOG4J_CONFIGURATION_FILE: /var/lib/besu/log4j.xml
+      JAVA_OPTS: -Xmx1536m -XX:+UnlockExperimentalVMOptions -XX:-UseG1GC -XX:+UseZGC
```
Comment (jonesho) [inline docker/compose.yml:144]: @Filter94 reduced this from 2GB to 1.5GB, tested it locally and should be safe for max heap size of l2-node-besu

Comment (Filter94) [reply docker/compose.yml:144]: Thanks! Let's see if it passes

**Review by jonesho — COMMENTED** [review]
@Filter94 I think we'd need to remove the `JAVA_OPTS` overrides in `docker/compose-local-dev.overrides.yml` as well

**Comment by Filter94** [comment]
@jonesho Done, thanks!

**Review by jonesho — APPROVED** [review]

File: docker/compose-local-dev.overrides.yml [hunk 2]
```diff
@@ -36,19 +36,7 @@ services:
...
-      JAVA_OPTS: -XX:+UnlockExperimentalVMOptions -XX:-UseG1GC -XX:+UseZGC
-
-  l1-el-node:
-    environment:
-      JAVA_OPTS: -XX:+UnlockExperimentalVMOptions -XX:-UseG1GC -XX:+UseZGC
   
   l2-node-besu:
     environment:
-      JAVA_OPTS: -XX:+UnlockExperimentalVMOptions -XX:-UseG1GC -XX:+UseZGC
+      JAVA_OPTS: -XX:+UnlockExperimentalVMOptions -XX:-UseG1GC -XX:+UseZGC
```
Comment (jonesho) [inline docker/compose-local-dev.overrides.yml:42]: @Filter94 I think we might need to keep the above overrides for `l2-node-besu` for now given that the pipeline was failing at the e2e geth test and I see local e2e geth test would still fail with OOM for max heap size set as `-Xmx2g`, provided that e2e geth test would be removed in the future, maybe we could keep this overrides for now, and in parallel, I'd search for a less heap size consumption alternative for transaction exclusion e2e test

**Review by jpnovais — APPROVED** [review]

**Review by VGau — APPROVED** [review]

**Pull request merged**